### DISCUSSION TITLE
How can I add shadows to layers?

### DISCUSSION THREAD

**Discussion by panosdk** [tannerhelland/PhotoDemon#587] [Q&A] [answered]
Cant find this option. Is it possible?

**Comment by hi5** [comment, marked as answer]
As far as I know there isn't a built-in tool so you will have to do it manually or record a macro.
Steps involved:

1. Duplicate layer
2. `Adjustments, Black & White` - be sure to have the B&W layer beneath the layer you wish to have a dropshadow.
3. `Effects, Blur` (play around with the type, settings)
4. You can save these steps as a macro (see `Tools` menu)

You can download my version of a drop shadow macro below, unpack it and copy/move it to `PhotoDemon\Data\Macros\` you can use the `Tools, Play Macro` or type it in the search bar for easy access (you may need to restart PD in case typing "drop" doesn't show it). 
![search](https://github.com/user-attachments/assets/70fe3188-4ef9-4239-806f-66e0f7b7ee1f)

Be sure your current / active layer is the one which you want to have a drop shadow.

📁 [dropshadow2.zip](https://github.com/user-attachments/files/16739370/dropshadow2.zip)

**Comment by panosdk** [comment]
Great thank you.
